master/internal/user/postgres_users.go

```diff
@@ -382,6 +388,21 @@ func ByToken(ctx context.Context, token string, ext *model.ExternalSessions) (
 	}
 
 	if session.Expiry.Before(time.Now()) {
+		var isRemote bool
+		if err := db.Bun().NewSelect().
+			Model(&model.User{}).
+			Column("remote").
+			Where("id = ?", session.UserID).
+			Scan(ctx, &isRemote); err != nil {
+			return nil, nil, db.ErrNotFound
+		}
+
+		// flag remote users as expired, so they are redirected to the SSO login page
+		// instead of returning an error
+		if isRemote {
+			return nil, nil, ErrRemoteUserTokenExpired
```

Comment:
Since this function's API now includes a particular error that callers can respond to (`ErrRemoteUserTokenExpired`), I like to include that in the doc comment. Something like
```go
// ByToken returns a user session given an authentication token. If a session is found but has
// expired, ErrRemoteUserTokenExpired will be returned.
```

Reply:
Sounds good! I'll make that adjustment!